HousingSearchApi.Tests/V1/Helper/ElasticSearchFixture.cs

```diff
@@ -1,37 +1,96 @@
-using Microsoft.Extensions.DependencyInjection;
+using HousingSearchApi.V1.Gateways.Models;
 using Nest;
 using System;
+using System.Collections.Generic;
+using System.Net.Http;
 using System.Threading;
 
 namespace HousingSearchApi.Tests.V1.Helper
 {
     public class ElasticSearchFixture : IDisposable
     {
+        public IElasticClient ElasticSearchClient => _factory?.ElasticSearchClient;
+        public HttpClient Client { get; private set; }
+
         private readonly MockWebApplicationFactory<Startup> _factory;
+        private static string _esAddress;
+
+        public List<QueryablePerson> Persons { get; private set; }
+        private readonly List<Action> _cleanup = new List<Action>();
 
         public ElasticSearchFixture()
         {
+            EnsureEnvVarConfigured("ELASTICSEARCH_DOMAIN_URL", "http://localhost:9200");
+
             _factory = new MockWebApplicationFactory<Startup>();
-            TestDataHelper.InsertPersonInEs(_factory.Services.GetService<IElasticClient>());
+            Client = _factory.CreateClient();
 
-            // For the index to have time to be populated
-            Thread.Sleep(500);
+            WaitForESInstance(ElasticSearchClient);
+            Persons = TestDataHelper.InsertPersonsInEs(ElasticSearchClient);
+
+            _cleanup.Add(() =>
+            {
+                ElasticSearchClient.DeleteManyAsync(Persons, TestDataHelper.Index)
+                                   .GetAwaiter().GetResult();
+            });
         }
 
         public void Dispose()
         {
             Dispose(true);
+            GC.SuppressFinalize(this);
         }
 
         private bool _disposed;
         protected virtual void Dispose(bool disposing)
         {
             if (disposing && !_disposed)
             {
+                foreach (var action in _cleanup)
+                    action();
+
                 if (null != _factory)
                     _factory.Dispose();
                 _disposed = true;
             }
         }
+
+        private void WaitForESInstance(IElasticClient elasticSearchClient)
+        {
+            Exception ex = null;
+            var timeout = DateTime.UtcNow.AddSeconds(10); // 5 second timeout (make configurable?)
```

Comment:
I don't think it's worth making configurable. But perhaps adding a comment to explain why we have this delay to begin with.

Reply:
Updated comment